example/client.rb

```diff
@@ -397,26 +414,38 @@ def webhook_parse_example
       }
     JSON
 
+    webhook_content = ResponseStruct.new(
+      headers,
+      body,
+      "<url>"
+    )
+
     webhook_api = Bitpesa::ApiClient.new
-    webhook = webhook_api.parse_response(webhook_content, "Webhook")
-
-    if webhook['event'].start_with?('transaction')
-      transaction_webhook = webhook_api.parse_response(webhook_content, 'TransactionWebhook')
-      puts transaction_webhook.object.to_s
-    elsif webhook['event'].start_with?('recipient')
-      recipient_webhook = webhook_api.parse_response(webhook_content, 'RecipientWebhook')
-      puts recipient_webhook.object.to_s
-    elsif webhook['event'].start_with?('payout_method')
-      payout_method_webhook = webhook_api.parse_response(webhook_content, 'PayoutMethodWebhook')
-      puts payout_method_webhook.object.to_s
-    elsif webhook['event'].start_with?('sender')
-      sender_webhook = webhook_api.parse_response(webhook_content, 'SenderWebhook')
-      puts sender_webhook.object.to_s
-    elsif webhook['event'].start_with?('document')
-      document_webhook = webhook_api.parse_response(webhook_content, 'DocumentWebhook')
-      puts document_webhook.object.to_s
-    else
-      puts webhook
+    webhook = webhook_api.parse_response(webhook_content.body, "Webhook")
+    nonce = JSON.parse(webhook_content.headers)['Authorization-Nonce']
+    signature = JSON.parse(webhook_content.headers)['Authorization-Signature']
+    webhook_url = webhook_content.url
+    body = webhook_content.body
+
+    if webhook_api.validate_webhook_request(webhook_url, body, nonce, signature)
```

Comment:
we might need to test as well that they key we got is the same key we store

Reply:
Also where was this `validate_webhook_request` method defined?

Reply:
Hey Zsolt I have been pairing with Linnie on this - the method `validate_webhook_request` is contained in api-specification project. Here is the link of the PR: [API-SPECIFICATION](https://github.com/bitpesa/bitpesa-api-specification/pull/99)